File: 268-MissingNumber/268-MissingNumber.java
// Last updated: 16/10/2025, 04:27:25
class Solution {
    public int missingNumber(int[] nums) {
        int n=nums.length;
        int ans = 0;
        for(int i=0;i<n;i++){
            ans ^= nums[i] ^ i+1;
        }
        return ans ;
    }
}

File: 268-MissingNumber/268-MissingNumber.cpp
// Last updated: 16/10/2025, 04:27:34
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int reqSum = n*(n+1)/2;
        int actualSum = 0;
        for(int i=0;i<n;i++){
            actualSum += nums[i];
        }
        int missing_number = reqSum - actualSum;
        
        return missing_number;
    }
};